controllers/productController.js

```diff
@@ -0,0 +1,48 @@
+const productService = require('../services/productService');
+
+const getProducts = async (req, res) => {
+  const limitValue = req.query.limit;
+
+  const result = await productService.getProducts(limitValue);
+
+  return res.status(200).json({ data: result });
+};
+
+const getProductsByMainCategory = async (req, res) => {
+  const { mainCategoryId } = req.params;
+
+  const result = await productService.getProductsByMainCategory(mainCategoryId);
+
+  return res.status(200).json({ data: result });
+};
+
+const getProductsBySubCategory = async (req, res) => {
+  const { mainCategoryId, subCategoryId } = req.params;
+  const formulation = req.query.formulation;
+  const scent = req.query.scent;
+  const limitValue = req.query.limit;
```

Comment:
구조분해할당을 사용하면 코드를 줄일 수 있습니다.
페이지네이션 구현을 위해 limit 값을 받아오는데 새로운 변수에 할당하지 말고 바로 사용해도 좋습니다.
```suggestion
  const { formulation, scent, limit } = req.query;
```